src/util/partition.hpp

```diff
@@ -171,6 +172,32 @@ make_partition(Part& divisions, const Sizes& sizes, T from=T{}) {
     return partition_view(divisions);
 }
 
+struct partition_functional_t {
+    constexpr partition_functional_t() {}
+};
+constexpr partition_functional_t partition_functional;
+
+template <
+    typename Part,
+    typename Range,
+    typename Func,
+    typename T = typename sequence_traits<Part>::value_type
+>
+partition_range<typename sequence_traits<Part>::const_iterator>
+make_partition(partition_functional_t,
```

Comment:
Yes, I think `transform_view` is appropriate here.
Specifically, `transform_view(make_span(0, n), ... your lambda/functor here ...)`.